alibi/datasets/__init__.py

```diff
@@ -0,0 +1,10 @@
+from alibi.utils.missing_optional_dependency import import_optional
+from .default import fetch_adult, fetch_imagenet, fetch_movie_sentiment, load_cats
+
+fetch_fashion_mnist = import_optional('alibi.datasets.tensorflow', names=['fetch_fashion_mnist'])
```

Comment:
@mauicv just to check my understanding. Has most of `alibi/datasets.py` been moved to `alibi/datasets/default.py` because of this line? i.e. we do this whenever we need to import something optional (in this case `fetch_fashion_mnist`) and add to `__all__`?

Reply:
Yeah exactly! If a feature needs an optional dependency then that feature is siloed with those imports into a separate file, in the above case `alibi/datasets/tensorflow.py`. Then we make sure to import that feature using `import_optional`. This is the pattern I've tried to follow throughout.

Reply:
Great, I think we should document this in developer docs too, i.e. all public API managed through `__init__` files with import guards for optional dependenies. The general pattern:

```
alibi.supackage
  __init__.py  # expose public API with optional import guards
 defaults.py  # private implementations requiring only core deps
 optional_dep.py # private implementations requiring an optional dependency (or several?)
```